### PR TITLE
[CPDLP-2468] UCL Deduping: Surface new deduped data in participants A…

### PR DESCRIPTION
**Pull request by mooktakim**
### Context

- Ticket: https://dfedigital.atlassian.net/browse/CPDLP-2468

### Changes proposed in this pull request

* Updated `Api::V3::ECF::ParticipantSerializer` to include `participant_id_changes`
* Updated `Api::V3::NPQParticipantSerializer` to include `participant_id_changes`
* Updated spec docs to make `participant_id_changes` a requirement
* New filter `from_participant_id` for ECF/NPQ participants endpoint
* Added feature flag `participant_id_changes`
* Updated release notes

### CODE REVIEW

File: app/serializers/api/v3/ecf/participant_serializer.rb [hunk 1]
```diff
@@ -141,6 +141,16 @@ def latest_induction_record(user:, profile:, lead_provider:)
             }
           }.compact
         end
+
+        attribute(:participant_id_changes, if: -> { FeatureFlag.active?(:participant_id_changes) } ) do |object|
+          (object.participant_id_changes || []).map do |c|
```
Comment (cwrw) [inline app/serializers/api/v3/ecf/participant_serializer.rb:146]: could we order by `created_at`?

File: swagger/v3/component_schemas/ECFParticipantAttributes.yml [hunk 2]
```diff
@@ -4,6 +4,7 @@ required:
   - full_name
   - updated_at
   - ecf_enrolments
+  - participant_id_changes
```
Comment (cwrw) [inline swagger/v3/component_schemas/ECFParticipantAttributes.yml:7]: could you add a new object participant id changes with the fields/explanations/examples to the spec please?

File: app/serializers/api/v3/ecf/participant_serializer.rb (hunk 1 above)
Comment (mooktakim) [reply app/serializers/api/v3/ecf/participant_serializer.rb:146]: I've added sort on the association:
https://github.com/DFE-Digital/early-careers-framework/pull/4138/files#diff-9802ca3c9c4cf89904fd44bc114e35ebdf2c5dd3d5b645491e2b253e1afef29bR42

**Review by leandroalemao — APPROVED** [review]
lgtm 👍 😄 tks

File: app/services/api/v3/ecf/participants_query.rb [hunk 3]
```diff
@@ -20,10 +20,12 @@ def participants_for_pagination
                     .select("users.id", "users.created_at", "users.updated_at")
                     .joins(participant_profiles: :induction_records)
                     .joins("JOIN (#{latest_induction_records_join.to_sql}) AS latest_induction_records_join ON latest_induction_records_join.latest_id = induction_records.id")
+                    .includes(:participant_id_changes)
```
Comment (cwrw) [inline app/services/api/v3/ecf/participants_query.rb:23]: maybe we can join rather than include at this stage?

**Review by cwrw — APPROVED** [review]

**Pull request merged**